File: Python3/201 - 400/242 Valid Anagram.py
# 44ms, 100.0%
class Solution:
    def isAnagram(self, s, t):
        if len(s) != len(t): return False
        s_set, t_set = set(s), set(t)
        if len(s_set) != len(t_set): return False
        for c in s_set:
            if s.count(c) != t.count(c): return False
        return True
    

File: 201 - 400/242 Valid Anagram.cc
// 22ms, 26.43%
class Solution
{
public:
    bool isAnagram(string s, string t)
    {
        unordered_map<char, int> tmp;
        for(auto x : s)
            if(tmp.find(x) == tmp.end()) tmp[x] = 1;
            else tmp[x] += 1;
        for(auto y : t)
            if(tmp.find(y) == tmp.end()) return false;
            else if(tmp[y] > 1) tmp[y] -= 1;
            else tmp.erase(y);
        return tmp.empty();
    }
};
